### ISSUE TITLE
Full Enigma support

### ISSUE THREAD

**Issue by chibill** [LapisBlue/Nocturne#22]

**Comment by caseif** [comment]
I don't really see a use case for this outside of using Nocturne and Enigma in parallel, which doesn't make sense since Nocturne is meant to be a successor to Enigma. Can you expand on why this feature might be needed?

**Comment by chibill** [comment]
Because Enigma mappings are neater. And also because Enigma allows you to bring mappings forward into newer versions of minecraft or what ever you are deobfuscating.

**Comment by jamierocks** [comment]
I'd sooner someone port the converter to SRG mappings, or to Nocturne than support Enigma mappings for export IMO.

**Comment by chibill** [comment]
Well that would be more difficult then just including enigma mappings... (He has his own odd stuff.)

**Comment by caseif** [comment]
This issue requires #10 to be fulfilled, which I'll personally be addressing.

**Comment by caseif** [comment]
`feature/enigma` has been merged into `develop`, so I'm closing this.

**Comment by chibill** [comment]
Currently if you say load SRG mappings then export them as enigma the resulting mappings can not be loaded.
`java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at javafx.fxml.FXMLLoader$MethodHandler.invoke(FXMLLoader.java:1770)
	at javafx.fxml.FXMLLoader$ControllerMethodEventHandler.handle(FXMLLoader.java:1653)
	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:49)
	at javafx.event.Event.fireEvent(Event.java:198)
	at javafx.scene.control.MenuItem.fire(MenuItem.java:462)
	at com.sun.javafx.scene.control.skin.ContextMenuContent$MenuItemContainer.doSelect(ContextMenuContent.java:1405)
	at com.sun.javafx.scene.control.skin.ContextMenuContent$MenuItemContainer.lambda$createChildren$341(ContextMenuContent.java:1358)
	at com.sun.javafx.scene.control.skin.ContextMenuContent$MenuItemContainer$$Lambda$320/796808022.handle(Unknown Source)
	at com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:218)
	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.event.Event.fireEvent(Event.java:198)
	at javafx.scene.Scene$MouseHandler.process(Scene.java:3758)
	at javafx.scene.Scene$MouseHandler.access$1500(Scene.java:3486)
	at javafx.scene.Scene.impl_processMouseEvent(Scene.java:1762)
	at javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2495)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:350)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:275)
	at java.security.AccessController.doPrivileged(Native Method)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$350(GlassViewEventHandler.java:385)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$$Lambda$297/1093952039.get(Unknown Source)
	at com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:404)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:384)
	at com.sun.glass.ui.View.handleMouseEvent(View.java:555)
	at com.sun.glass.ui.View.notifyMouse(View.java:927)
	at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at com.sun.glass.ui.win.WinApplication.lambda$null$145(WinApplication.java:101)
	at com.sun.glass.ui.win.WinApplication$$Lambda$36/1963387170.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at sun.reflect.misc.Trampoline.invoke(MethodUtil.java:71)
	at sun.reflect.GeneratedMethodAccessor1.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at sun.reflect.misc.MethodUtil.invoke(MethodUtil.java:275)
	at javafx.fxml.FXMLLoader$MethodHandler.invoke(FXMLLoader.java:1765)
	... 46 more
Caused by: java.lang.NullPointerException
	at blue.lapis.nocturne.jar.model.JarClassEntry.getDeobfuscatedName(JarClassEntry.java:119)
	at blue.lapis.nocturne.jar.model.hierarchy.Hierarchy.fromSet(Hierarchy.java:52)
	at blue.lapis.nocturne.jar.model.ClassSet.generateHierarchy(ClassSet.java:152)
	at blue.lapis.nocturne.jar.model.ClassSet.getDeobfuscatedHierarchy(ClassSet.java:140)
	at blue.lapis.nocturne.gui.MainController.updateDeobfuscatedClassListView(MainController.java:279)
	at blue.lapis.nocturne.gui.MainController.updateClassViews(MainController.java:321)
	at blue.lapis.nocturne.mapping.model.ClassMapping.setDeobfuscatedName(ClassMapping.java:200)
	at blue.lapis.nocturne.mapping.model.TopLevelClassMapping.setDeobfuscatedName(TopLevelClassMapping.java:86)
	at blue.lapis.nocturne.mapping.model.TopLevelClassMapping.setDeobfuscatedName(TopLevelClassMapping.java:81)
	at blue.lapis.nocturne.util.helper.MappingsHelper.genClassMapping(MappingsHelper.java:112)
	at blue.lapis.nocturne.mapping.io.reader.EnigmaReader.read(EnigmaReader.java:111)
	at blue.lapis.nocturne.gui.io.mappings.MappingsOpenDialogHelper.openMappings(MappingsOpenDialogHelper.java:83)
	at blue.lapis.nocturne.gui.MainController.loadMappings(MainController.java:198)
	... 56 more
`

**Comment by chibill** [comment]
Just so you know that its still buggy.
